[DOC] Adds tls-external to user authentication concepts (#5294)
strimzi/strimzi-kafka-operator: documentation/modules/con-securing-client-authentication.adoc

[id='con-securing-client-authentication-{context}']
= User authentication

[role="_abstract"]
User authentication is configured using the `authentication` property in `KafkaUser.spec`.
The authentication mechanism enabled for the user is specified using the `type` field.

Supported authentication types:

* `tls` for TLS client authentication
* `tls-external` for TLS client authentication using external certificates
* `scram-sha-512` for SCRAM-SHA-512 authentication

If `tls` or `scram-sha-512` is specified, the User Operator creates authentication credentials when it creates the user.

> If `tls-external` is specified, a TLS user is created without authentication credentials.

Reviewer: I think this is confusing. _Creating the TLS user_ is generating the certificate / authentication credentials. So you cannot really say _user is created_ at the same time as _without authentication credentials_. Maybe the below would be more accurate.

```suggestion
If `tls-external` is specified, the user is expected to use TLS client authentication, but no authentication credentials are created.
```

When no authentication type is specified, the User Operator does not create the user or its credentials.

Use `tls-external` to authenticate with TLS using a certificate issued outside the User Operator.

> The User Operator creates a TLS user, but it does not generate a TLS certificate or a secret.

Reviewer: Same as above, it does not really create the user because the user is created just by creating the certificate.

Author: Changed to: _The User Operator does not generate a TLS certificate or a secret._

> You can still manage user authorization through the User Operator as a TLS user.

Reviewer 2: standard TLS user?

I'm not sure of the correct terminology. Maybe "TLS user" is clear enough.

Author: I added _standard_ then took it out. 
I'll add it again and see if Jakub thinks we don't need it...

  user.password: # Password for protecting the PKCS #12 archive file
----

> == TLS client authentication using an external certificate

Reviewer: I think _external certificate_ might be confusing. It still needs to be a certificate signed by the Clients CA. It is just expected to be signed by someone / something else than the User Operator. I guess it is hard to put it into title. But I'm afraid the current title would confuse people into thinking that they can use this to bring their own custom certificate with this which is not through. It is just _issued outside of User Operator_.

Author: We can change to that then: _TLS client authentication using a certificate issued outside the User Operator_

To use TLS client authentication using a certificate issued outside the User Operator, you set the `type` field in the `KafkaUser` resource to `tls-external`.
A secret and credentials are not created for the user.

> .Example user with TLS client authentication using external certificates enabled

Reviewer: Same as above.

Author: Changed to: _.Example user with TLS client authentication that uses a certificate issued outside the User Operator_

  user.password: # Password for protecting the PKCS #12 archive file
----

== TLS client authentication using an external certificate

> To use TLS client authentication using a certificate issued outside the User Operator, you set the `type` field to `tls-external`.

Reviewer 2: ```suggestion
To use TLS client authentication using a certificate issued outside the User Operator, you set the `type` field in the `KafkaUser` resource to `tls-external`.
```

Author: I added for all three mechanisms

A secret and credentials are not created for the user.

> .An example `KafkaUser` with TLS client authentication enabled

Reviewer 2: ```suggestion
.An example `KafkaUser` with TLS client authentication using external certificates enabled
```

[source,yaml,subs="attributes+"]
----
apiVersion: {KafkaUserApiVersion}
kind: KafkaUser
metadata:
  name: my-user
  labels:
    strimzi.io/cluster: my-cluster
spec:
  authentication:
    type: tls-external
  # ...
----

> 

Reviewer 2: Remove extra space.

.Additional resources

> 

Reviewer 2: Missing additional resources.

Author: I moved them, but missed the title. Removed.
